src/main/java/net/dries007/tfc/common/blockentities/BloomeryBlockEntity.java

```diff
@@ -0,0 +1,443 @@
+                bloomery.cachedRecipe = null; // Clear recipe
+
+                bloomery.updateMoltenBlock(false);
+                state = state.setValue(BloomeryBlock.LIT, false);
+                level.setBlockAndUpdate(pos, state);
+            }
+
+            // Update multiblock status
+            int newMaxItems = BloomeryBlock.getChimneyLevels(level, bloomery.getInternalBlock()) * 8;
+            Direction direction = state.getValue(BloomeryBlock.FACING);
+
+            //todo: abort process/dump items if !isFormed?
+            if (!BloomeryBlock.isFormed(level, bloomery.getInternalBlock(), direction))
+            {
+                newMaxItems = 0;
+            }
+
+            bloomery.maxInput = newMaxItems;
+            // since amount of catalyst no longer strictly equals number of iron ores/input, maxCatalyst is determined by amount needed to satisfy current amount of input in the bloomery (maxed at 128 just for sanity's sake)
+            if (bloomery.cachedRecipe == null)
+            {
+                bloomery.maxCatalyst = 0;
+            }
+            else
+            {
+                int maxCatalyst = bloomery.getTotalInput().getAmount() / bloomery.cachedRecipe.getInputFluid().getAmount() * bloomery.cachedRecipe.getCatalyst().getCount();
+                bloomery.maxCatalyst = Math.min(maxCatalyst, 128);
+            }
+
+            boolean turnOff = false;
+            while (bloomery.maxInput < bloomery.inputStacks.size())
+            {
+                turnOff = true;
+                // Structure lost one or more chimney levels
+                Helpers.spawnItem(level, bloomery.worldPosition, bloomery.inputStacks.get(0));
+                bloomery.inputStacks.remove(0);
+                bloomery.markForSync();
+            }
+            while (bloomery.maxCatalyst < bloomery.catalystStacks.size())
+            {
+                turnOff = true;
+                Helpers.spawnItem(level, bloomery.worldPosition, bloomery.catalystStacks.get(0));
+                bloomery.catalystStacks.remove(0);
+                bloomery.markForSync();
+            }
+            // Structure became compromised, unlit if needed
+            if (turnOff && state.getValue(BloomeryBlock.LIT))
+            {
+                state = state.setValue(BloomeryBlock.LIT, false);
+                level.setBlockAndUpdate(pos, state);
+            }
+            if (!BloomeryBlock.canGateStayInPlace(level, pos, direction.getAxis()))
+            {
+                // Bloomery gate (the front facing) structure became compromised
+                level.destroyBlock(pos, true);
+                return;
+            }
+
+//            if (!BloomeryBlock.isFormed(level, bloomery.getInternalBlock(), state.getValue(BloomeryBlock.FACING)) || )
+//            if (!bloomery.isInternalBlockComplete() && !bloomery.catalystStacks.isEmpty())
+//            {
+//                bloomery.dumpItems();
+//            }
```

Comment:
Before we merge you need to decide on stuff like this. We generally don't leave commented code in